components/raftstore/src/store/fsm/store.rs

```diff
@@ -2809,15 +2764,22 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
         {
             let meta = self.ctx.store_meta.lock().unwrap();
             for sst in ssts {
-                if let Some(r) = meta.regions.get(&sst.get_region_id()) {
+                if sst.api_version < sst_importer::API_VERSION_2 {
+                    // SST of old versions are created by old TiKV and have different prerequisite
+                    // we can't delete them here. They can only be deleted manually
```

Comment:
Is it better to keep the older clean logic here for old SST? Otherwise it may lead to too many support tickets after the upgrade.

Reply:
I'm worried about the previous panic problem which is caused by this cleanup. 

To make it clear, the affected case is 
- when lightning local backend is importing data while upgrading or fast adding index is running while upgrading. 
- and the ingest RPC does not come so TiKV has no chance to trigger check_sst_for_ingestion then delete the stale SST

I think the affected case is not often happened, so it's not a big problem if disk is occupied by old version SST files. But panic is too severe.

Reply:
The panic rarely happens but the old SST files are sure to happen be there forever, which will be challenged by customer for sure. 
Another way is that we can delete old SST files after a safe time, saying 1 week. So that they can be eventually deleted instead of stay there forever.

Reply:
https://github.com/tikv/tikv/pull/15795/commits/d95c7fc35dfbbf65ce2dd91ed02bd1350a776604 

When the tikv has that region, we check the staleness by region epoch. When it's not the case, for api_version_2 we can safely delete the files, for api_version_1 we check last_modified time